crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -705,7 +705,9 @@ impl<'db> TypeInferenceBuilder<'db> {
         }
 
         let function_type = FunctionType::new(self.db, name.id.clone(), definition, decorator_tys);
-        let function_ty = if function_type.is_stdlib_symbol(self.db, "typing", "reveal_type") {
+        let function_ty = if function_type.is_stdlib_symbol(self.db, "typing", "reveal_type")
+            || function_type.is_stdlib_symbol(self.db, "typing_extensions", "reveal_type")
+        {
```

Comment:
This solves the specific case, but not the general case. In general, there'll be a large number of symbols that could come from `typing` or `typing_extension`, and we won't care which. I think we should add a separate method to `FunctionType` (`FunctionType::is_typing_symbol`?) where `foo.is_typing_symbol("X")` returns `true` if the qualified name of the type of `foo` is either `typing.X` or `typing_extensions.X`. That will make it less likely (though of course not impossible) in the future that we'll forget to account for the possibility that the symbol might come from `typing_extensions` rather than `typing`